python/lsst/ts/atmcssimulator/mcs_csc.py

```diff
@@ -77,973 +51,267 @@ def __init__(self, initial_state: salobj.State = salobj.State.STANDBY) -> None:
-                    f"start_position[{axis}]={start_position[axis]}"
-                )
-            if max_commanded_position[axis] < start_position[axis]:
-                raise salobj.ExpectedError(
-                    f"max_commanded_position[{axis}]={max_commanded_position[axis]} < "
-                    f"start_position[{axis}]={start_position[axis]}"
-                )
-
-        if max_velocity.min() <= 0:
-            raise salobj.ExpectedError(
-                f"max_velocity={max_velocity}; all values must be positive"
-            )
-        if max_acceleration.min() <= 0:
-            raise salobj.ExpectedError(
-                f"max_acceleration={max_acceleration}; all values must be positive"
-            )
-        if limit_overtravel < 0:
-            raise salobj.ExpectedError(
-                f"limit_overtravel={limit_overtravel} must be >= 0"
-            )
-
-        self.max_tracking_interval = max_tracking_interval
-        self.min_commanded_position = min_commanded_position
-        self.max_commanded_position = max_commanded_position
-        self.min_limit_switch_position = min_limit_switch_position
-        self.max_limit_switch_position = max_limit_switch_position
-        self.max_velocity = max_velocity
-        self.topple_azimuth = topple_azimuth
-        self.m3_port_positions = m3_port_positions
-        self.axis_encoder_counts_per_deg = axis_encoder_counts_per_deg
-        self.motor_encoder_counts_per_deg = motor_encoder_counts_per_deg
-        self.motor_axis_ratio = motor_axis_ratio
-        self.torque_per_accel = torque_per_accel
-        self.nsettle = nsettle
-        self.limit_overtravel = limit_overtravel
-
-        tai = utils.current_tai()
-        self.actuators = [
-            simactuators.TrackingActuator(
-                min_position=self.min_commanded_position[axis],
-                max_position=self.max_commanded_position[axis],
-                max_velocity=max_velocity[axis],
-                max_acceleration=max_acceleration[axis],
-                # Use 0 for M3 to prevent tracking.
-                dtmax_track=0 if axis == 4 else self.max_tracking_interval,
-                nsettle=self.nsettle,
-                tai=tai,
-                start_position=start_position[axis],
-            )
-            for axis in Axis
-        ]
-        self.actuators[0].verbose = True
-
-        await self.evt_positionLimits.set_write(
-            minimum=min_commanded_position,
-            maximum=max_commanded_position,
-            force_output=True,
-        )
+        try:
+            await issued_command.done
+        except Exception:
+            self.log.exception(f"IssuedCommand {issued_command.name} failed.")
+            result = "not acknowledged"
+            if issued_command.ack.done():
+                result = "failed"
+            raise salobj.ExpectedError(f"Command do_startTracking {result}.")
```

Comment:
It really should be that simple, That is the main point I'm trying to make when I say that IssuedCommand should *guarantee* that "done" and "ack" are always both set done in some way when a command is finished.

That way you can do the thing I showed above.

IssuedCommand has a bit of extra logic to handle all the different ways a command can be done, so the user doesn't have to think about it.

Reply:
As agreed during our chat today on Slack, I removed the `ack` future completely. I also removed the monitor method since it now indeed isn't necessary anymore.